src/Database/PG/Query/Pool.hs

```diff
@@ -168,12 +166,11 @@ class FromPGTxErr e where
 class FromPGConnErr e where
   fromPGConnErr :: PGConnErr -> e
 
-runTxOnConn :: (MonadIO m, FromPGTxErr e)
-            => PGConn
-            -> TxMode
-            -> (PGConn -> ExceptT e m a)
-            -> ExceptT e m a
-runTxOnConn pgConn txm f = do
+asTransaction :: (MonadIO m, FromPGTxErr e)
+              => TxMode
+              -> (PGConn -> ExceptT e m a)
+              -> PGConn -> ExceptT e m a
```

Comment:
I think the rename is good, but I'm not sure about the change in ordering of inputs. The reason functions are usually last is to allow stuff like `asTransaction mode conn $ \c -> ...`

Reply:
So the reason for the reordering is that I found the type signature to be illustrative this way: given a transaction mode, we convert a "action on a conn" to a "action on a conn". (And the only place that it's used, in the definition of `runTx`, actually plays nicely with this.)